src/colorizer/state/slices/dataset_slice.ts

```diff
@@ -32,9 +32,10 @@ export const createDatasetSlice: StateCreator<CollectionSlice & DatasetSlice & B
     if (backdropKey === null || !dataset.hasBackdrop(backdropKey)) {
       backdropKey = dataset.getDefaultBackdropKey();
     }
+    let backdropVisible = get().backdropVisible && backdropKey !== null;
 
     // TODO: Dispose of old dataset?
-    set({ datasetKey: key, dataset, backdropKey });
+    set({ datasetKey: key, dataset, backdropKey, backdropVisible });
   },
-  clearDataset: () => set({ datasetKey: null, dataset: null, backdropKey: null }),
+  clearDataset: () => set({ datasetKey: null, dataset: null, backdropKey: null, backdropVisible: false }),
```

Comment:
Fix. This disables the backdrop visibility if the incoming dataset has no backdrops.

Reply:
- [ ] TODO: Test for this in the dataset_slice.ts unit tests